src/App.jsx

```diff
@@ -14,33 +17,103 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_APLHABET = 'alphabet';
+const SORT_LENGTH = 'length';
+const INITIAL_STATE = '';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const getPreparedGoods = (goods, action, reverse) => {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (action) {
+    switch (action) {
+      case SORT_APLHABET:
+        preparedGoods.sort();
+        break;
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+      case SORT_LENGTH:
+        preparedGoods.sort((good1, good2) => good1.length - good2.length);
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+};
+
+const GoodList = ({ goods }) => {
+  return goods.map(good => (
+    <li key={good} data-cy="Good">
+      {good}
+    </li>
+  ));
+};
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortAction, setSortAction] = useState(INITIAL_STATE);
+  const [isReversed, setIsReversed] = useState(false);
+  const showReset = sortAction || isReversed;
+  const resetHandler = () => {
```

Comment:
The method name should start with a verb. It is a common best practice to start function names with a verb because it describes the action the function performs. Consider renaming `resetHandler` to something like `handleReset`.